### PR TITLE
#2043 Adds changelogs for unreleased changes.

### PR DESCRIPTION
**Pull request by ashishkujoy**
I like the idea of maintaining a changelog for unreleased changes mentioned [here](https://keepachangelog.com/en/1.0.0/).
I hope this will help in future release activity and at the same time should give users of kotest a gist what all development is going on in Kotest.

I think I have covered all the changes from last release in case If I have missed then please add that.

### CODE REVIEW

File: documentation/docs/changelog.md [hunk 1]
```diff
@@ -5,6 +5,61 @@ slug: changelog.html
...
+* Added `kotest-extensions-wiremock` module for managing lifecycle of `WireMockServer` in Kotest test. #2108
+* Upgrade `klock` dependency to 2.0.6 and added `browser`, `nodejs`, `linuxX64`, `mingwX64`, `macosX64`, `tvos`,
+  `iosX64`, `iosArm64` and `iosArm32` platform targets for `kotest-assertions-klock`. #2116
+* Run eventually one final time if iterations is one and delay is greater than the interval #2105
+* Updates `withEnvironment` function to ignore environment variable key case sensitivity on Windows platform. #2099
+* Some improvement around `eventually`
+  1. Makes EventuallyPredicate<T> a type alias instead of interface for better user experiance.
+  2. Update failure message to inform the user about failure of given EventaullyPredicate.
+  3. Adds an overload of `eventually` which does not takes `EventuallyPredicate` so that it give feel
+     of `util` function. #2046
```
Comment (jschneidereit) [inline documentation/docs/changelog.md]: Typo: "until"

**Review by jschneidereit — APPROVED** [review]

**Comment by sksamuel** [comment]
Nice idea.

**Comment by sksamuel** [comment]
lgtm

File: documentation/docs/changelog.md [hunk 2]
```diff
@@ -5,6 +5,61 @@ slug: changelog.html
...
+
+
+
+#### BugFixes.
```
Comment (ajalt) [inline documentation/docs/changelog.md]:
```suggestion
#### Bugfixes
```

To match the format of the previous entries

Comment (ashishkujoy) [reply documentation/docs/changelog.md]: Corrected

**Pull request merged**